editor/quick_open_dialog.cpp

```diff
@@ -0,0 +1,980 @@
+		_select_item(p_index);
+		emit_signal(SNAME("result_clicked"));
+	}
+}
+
+void QuickOpenResultContainer::_toggle_include_addons(bool p_pressed) {
+	EditorSettings::get_singleton()->set("filesystem/quick_open_dialog/include_addons", p_pressed);
+
+	cleanup();
+	_create_initial_results(p_pressed);
+}
+
+void QuickOpenResultContainer::_toggle_display_mode() {
+	ContentDisplayMode new_display_mode = (content_display_mode == ContentDisplayMode::LIST) ? ContentDisplayMode::GRID : ContentDisplayMode::LIST;
+	_set_display_mode(new_display_mode);
+}
+
+void QuickOpenResultContainer::_set_display_mode(ContentDisplayMode p_display_mode) {
+	content_display_mode = p_display_mode;
+
+	const bool first_time = !list->is_visible() && !grid->is_visible();
+
+	if (!first_time) {
+		const bool show_list = (content_display_mode == ContentDisplayMode::LIST);
+		if ((show_list && list->is_visible()) || (!show_list && grid->is_visible())) {
+			return;
+		}
+	}
+
+	hide();
+
+	// Move result item nodes from one container to the other.
+	CanvasItem *prev_root;
+	CanvasItem *next_root;
+	if (content_display_mode == ContentDisplayMode::LIST) {
+		prev_root = Object::cast_to<CanvasItem>(grid);
+		next_root = Object::cast_to<CanvasItem>(list);
+	} else {
+		prev_root = Object::cast_to<CanvasItem>(list);
+		next_root = Object::cast_to<CanvasItem>(grid);
+	}
+
+	prev_root->hide();
+	for (QuickOpenResultItem *item : result_items) {
+		item->set_display_mode(content_display_mode);
+
+		if (!first_time) {
+			prev_root->remove_child(item);
+		}
+
+		next_root->add_child(item);
+	}
+	next_root->show();
+	show();
+
+	_update_result_items(num_visible_results, selection_index);
+
+	if (content_display_mode == ContentDisplayMode::LIST) {
+		display_mode_toggle->set_icon(get_editor_theme_icon(SNAME("FileThumbnail")));
+		display_mode_toggle->set_tooltip_text(TTR("Grid view"));
+	} else {
+		display_mode_toggle->set_icon(get_editor_theme_icon(SNAME("FileList")));
+		display_mode_toggle->set_tooltip_text(TTR("List view"));
+	}
+}
+
+bool QuickOpenResultContainer::has_nothing_selected() const {
+	return selection_index < 0;
+}
+
+String QuickOpenResultContainer::get_selected() const {
+	ERR_FAIL_COND_V_MSG(has_nothing_selected(), "", "Tried to get selected file, but nothing was selected.");
+
+	const Candidate &c = candidates[selection_index];
+	return c.file_directory.path_join(c.file_name);
+}
+
+void QuickOpenResultContainer::cleanup() {
+	num_visible_results = 0;
+	candidates.clear();
+	_select_item(-1);
+
+	for (QuickOpenResultItem *item : result_items) {
+		item->reset();
+	}
+}
+
+void QuickOpenResultContainer::_notification(int p_what) {
+	switch (p_what) {
+		case NOTIFICATION_ENTER_TREE:
+		case NOTIFICATION_THEME_CHANGED: {
+			Color text_color = get_theme_color(SNAME("readonly_color"), EditorStringName(Editor));
+			file_details_path->add_theme_color_override("font_color", text_color);
+			no_results_label->add_theme_color_override("font_color", text_color);
```

Comment:
You are using SNAME inconsistently. This code is ran only once (unless you change editor theme), so SNAME is not necessary here.